pkg/operators/wasm/cache.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2025 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wasm
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/tetratelabs/wazero"
+)
+
+// compilationCache is a cache for the wasm compilation based on wazero.CompilationCache.
+type compilationCache struct {
+	cache       wazero.CompilationCache
+	refreshTime time.Time
+	refreshMu   sync.Mutex
+}
+
+func newCompilationCache() *compilationCache {
+	return &compilationCache{
+		cache:       wazero.NewCompilationCache(),
+		refreshTime: time.Now(),
+	}
+}
+
+func (c *compilationCache) refreshCache() {
+	c.refreshMu.Lock()
+	defer c.refreshMu.Unlock()
+
+	if time.Since(c.refreshTime) > time.Hour {
+		if c.cache != nil {
+			c.cache.Close(context.Background())
```

Comment:
Can it happen that a gadget that is already running now has problems when calling guest functions? Or is the cache only used while initializing the module as a whole?

Reply:
> Or is the cache only used while initializing the module as a whole?

Yes, it is only used during initializing: https://github.com/inspektor-gadget/inspektor-gadget/blob/main/pkg/operators/wasm/wasm.go#L253